storage/innobase/dict/dict0mem.cc

```diff
@@ -1261,6 +1262,15 @@ bool dict_table_t::deserialise_columns(const byte* metadata, ulint len)
 	DBUG_ASSERT(col == &dropped_cols[n_dropped_cols]);
 
 	UT_LIST_GET_FIRST(indexes)->reconstruct_fields();
+
+	rw_lock_x_lock(&index->lock);
+	committed_count_inited = (12 + num_non_pk_fields * 2 == len);
+	if (committed_count_inited) {
```

Comment:
can the lock be placed after the condition? Or removed entirely here?  `committed_count` being an atomic type is already protected. Is there a concurrency issue around this function?

Reply:
The lock was intended to protect both `dict_table_t.committed_count_inited` and `dict_table_t.committed_count`. I don't remember the initial reason for why synchronization was needed but upon reviewing this, you're right.